tests/jax/components/training/step_test.py

```diff
@@ -0,0 +1,284 @@
+
+    def __init__(self) -> None:
+        trainer_agent_net_keys = {
+            "agent_0": "network_agent_0",
+            "agent_1": "network_agent_1",
+            "agent_2": "network_agent_2",
+        }
+        networks = {
+            "networks": {
+                "network_agent_0": SimpleNamespace(
+                    params={"key": jnp.array([0.0, 0.0, 0.0])},
+                    network=SimpleNamespace(apply=apply),
+                ),
+                "network_agent_1": SimpleNamespace(
+                    params={"key": jnp.array([1.0, 1.0, 1.0])},
+                    network=SimpleNamespace(apply=apply),
+                ),
+                "network_agent_2": SimpleNamespace(
+                    params={"key": jnp.array([2.0, 2.0, 2.0])},
+                    network=SimpleNamespace(apply=apply),
+                ),
+            }
+        }
+        opt_states = {
+            "network_agent_0": 0,
+            "network_agent_1": 1,
+            "network_agent_2": 2,
+        }
+        store = SimpleNamespace(
+            dataset_iterator=iter([1, 2, 3]),
+            step_fn=step_fn,
+            timestamp=1657703548.5225394,  # time.time() format
+            trainer_parameter_client=MockParameterClient(),
+            trainer_counts={"next_sample": 2},
+            trainer_logger=MockTrainerLogger(),
+            sample_batch_size=2,
+            sequence_length=3,
+            trainer_agent_net_keys=trainer_agent_net_keys,
+            networks=networks,
+            gae_fn=gae_advantages,
+            opt_states=opt_states,
+            key=jax.random.PRNGKey(5),
+            num_minibatches=1,
+            epoch_update_fn=epoch_update,
+            num_epochs=2,
+        )
+        self.store = store
+
+
+@pytest.fixture
+def mock_trainer() -> MockTrainer:
+    """Build fixture from MockTrainer"""
+    return MockTrainer()
+
+
+@pytest.fixture
+def dummy_sample() -> DummySample:
+    """Build fixture from DummySample"""
+    return DummySample()
+
+
+def test_default_trainer_step_initiator() -> None:
+    """Test constructor of DefaultTrainerStep component"""
+    trainer_step = DefaultTrainerStep()
+    assert trainer_step.config.random_key == 42
+
+
+def test_on_training_step_with_timestamp(mock_trainer: Trainer) -> None:
+    """Test on_training_step method from TrainerStep case of existing timestamp"""
+    trainer_step = DefaultTrainerStep()
+    old_timestamp = mock_trainer.store.timestamp
+    trainer_step.on_training_step(trainer=mock_trainer)
+
+    assert mock_trainer.store.timestamp > old_timestamp
+
+    assert list(mock_trainer.store.trainer_parameter_client.params.keys()) == [
+        "trainer_steps",
+        "trainer_walltime",
+    ]
+    assert mock_trainer.store.trainer_parameter_client.params["trainer_steps"] == 1
+    elapsed_time = int(time.time() - old_timestamp)
+    assert (
+        int(mock_trainer.store.trainer_parameter_client.params["trainer_walltime"])
+        == elapsed_time
```

Comment:
This might run into the same `time.time()` issue as the previous check could have. Is there another way of checking this as well?

Reply:
We can change the assert to check that is different of zero